src/loaders/OmeZarrLoader.ts

```diff
@@ -144,55 +154,103 @@ class OMEZarrLoader extends ThreadableVolumeLoader {
-    const { multiscales, omero } = group.attrs as OMEZarrMetadata;
-
-    // Pick scene (multiscale)
-    if (scene > multiscales.length) {
-      console.warn(`WARNING: OMEZarrLoader: scene ${scene} is invalid. Using scene 0.`);
-      scene = 0;
-    }
-    const multiscale = multiscales[scene];
-
-    // Open all scale levels of multiscale
-    const scaleLevelPromises = multiscale.datasets.map(({ path }) => zarr.open(root.resolve(path), { kind: "array" }));
-    const scaleLevels = (await Promise.all(scaleLevelPromises)) as NumericZarrArray[];
-    const axisTCZYX = remapAxesToTCZYX(multiscale.axes);
+    const urlsArr = Array.isArray(urls) ? urls : [urls];
+    const scenesArr = Array.isArray(scenes) ? scenes : [scenes];
+
+    // Create one `ZarrSource` per URL
+    const sourceProms = urlsArr.map(async (url, i) => {
+      const store = new WrappedStore<RequestInit>(new FetchStore(url), cache, queue);
+      const root = zarr.root(store);
+      const group = await zarr.open(root, { kind: "group" });
+      const { multiscales, omero } = group.attrs as OMEZarrMetadata;
+
+      // Pick scene (multiscale)
+      let scene = scenesArr[Math.min(i, scenesArr.length - 1)];
+      if (scene > multiscales.length) {
+        console.warn(`WARNING: OMEZarrLoader: scene ${scene} is invalid. Using scene 0.`);
+        scene = 0;
+      }
+      const multiscaleMetadata = multiscales[scene];
+
+      // Open all scale levels of multiscale
+      const lvlProms = multiscaleMetadata.datasets.map(({ path }) => zarr.open(root.resolve(path), { kind: "array" }));
+      const scaleLevels = (await Promise.all(lvlProms)) as NumericZarrArray[];
+      const axesTCZYX = remapAxesToTCZYX(multiscaleMetadata.axes);
+
+      return {
+        scaleLevels,
+        multiscaleMetadata,
+        omeroMetadata: omero,
+        axesTCZYX,
+        channelOffset: 0,
+      } as ZarrSource;
+    });
+    const sources = await Promise.all(sourceProms);
 
+    // Set `channelOffset`s so we can match channel indices to sources
+    let channelCount = 0;
+    for (const s of sources) {
+      s.channelOffset = channelCount;
+      channelCount += s.omeroMetadata.channels.length;
+    }
+    // Ensure the sizes of all sources' scale levels are matched up. See this function's docs for more.
+    matchSourceScaleLevels(sources);
+    // TODO: if `matchSourceScaleLevels` returned successfully, every one of these sources' `multiscaleMetadata` is the
+    // same in every field we care about, so we only ever use the first source's `multiscaleMetadata` after this point.
+    // Should we only store one `OMEMultiscale` record total, rather than one per source?
     const priorityDirs = fetchOptions?.priorityDirections ? fetchOptions.priorityDirections.slice() : undefined;
-    return new OMEZarrLoader(store, scaleLevels, multiscale, omero, axisTCZYX, queue, fetchOptions, priorityDirs);
+    return new OMEZarrLoader(sources, queue, fetchOptions, priorityDirs);
```

Comment:
The "private constructor" pattern I have going on here is strange as well, but I blame that mostly on JavaScript. It's a consequence of having both classes with privileged constructors and `async`, and not anticipating how they might interact (which is pretty reasonable). I feel okay about this both because I kind of just treat the constructor as the final step in this static method, and because all the *dimensions* (and metadata) loading happens here, and all the *data* loading happens in the rest of the class.